feat(mobile): Search Page

YavizGuldalf opened the pull request:
Fixes #465
This PR includes the designs for the search page. This search page is designed to be a general search for the app, showing search results for users, quizzes and forums. This is up-for-debate though, as we already have search functionality in the Quiz and Forum pages separately. Still, having a general place for searching may be nice.

 I have also added sorting and filtering options. You can sort by newest or most liked, and you can use the filters to constrain the search to certain content types and tags. The search functionality was left entirely to the backend. I have integrated the page's design with quizCard and userCard components, but forum post and comment cards have not been merged to main yet and therefore they have placeholders instead. You can find example screenshot below.
 
 
<img width="230" src="https://github.com/user-attachments/assets/596789f4-b11c-4acb-8598-03ce57904d1a">
<img width="230" src="https://github.com/user-attachments/assets/90169305-b883-4edb-bb10-99e23dfdd561">
<img width="230" src="https://github.com/user-attachments/assets/5fb2c362-b6fe-4048-a8a1-85fc3587e049">
<img width="230" src="https://github.com/user-attachments/assets/1873b308-5bea-4a25-b6d6-6cd8df8c9746">
<img width="230" src="https://github.com/user-attachments/assets/d8a71f8a-416f-4971-a73a-1e1fcb341ea5">
<img width="230" src="https://github.com/user-attachments/assets/67eacf94-4193-4985-ac80-93a8320a1402">

aoengin commented:
<img width="284" alt="Screenshot 2024-11-18 at 07 17 53" src="https://github.com/user-attachments/assets/f2a05015-4c51-4b67-827c-5a0bde51ca8d">

Can you fix the visual bug in the tags?

It would also be good if the 'Sort By' feature had an option to clear the selected choice.

YavizGuldalf commented:
@aoengin, I have added a few commits that should fix the issues you mentioned. I don't know the exact model of the virtual device you are using to simulate the program but I tested this new version on a medium sized phone (Pixel 4) and the results look like the following:

<img src="https://github.com/user-attachments/assets/c5236ac5-4e5c-42df-906a-dec1e3fe3742" width="230">

I have also added the option to clear the 'Sort by' feature: 

<img src="https://github.com/user-attachments/assets/619b3901-33b1-4bc4-9af5-72d9042ebce0" width="230">

Can you please review these changes again?

aoengin commented:
Dividing the options into "cards" would be a good idea since there are no borders between the choices currently. However, I'll merge it for now and we can improve the UI later.
